ihmc-graphics/src/libgdx/java/us/ihmc/rdx/vr/RDXVRManager.java

```diff
@@ -99,110 +79,81 @@ private boolean pollEvents(RDXBaseUI baseUI)
          {
             initializing = true;
             contextCreatedNotification = new Notification();
-            ThreadTools.startAsDaemon(() ->
-            {
-               synchronized (syncObject)
-               {
-                  initSystemCount++;
-                  context.initSystem();
-               }
-               contextCreatedNotification.set();
-            }, DefaultExceptionHandler.MESSAGE_AND_STACKTRACE, getClass().getSimpleName() + "-initSystem");
+            context.initSystem(); // May block for a bit
+            contextCreatedNotification.set();
          }
          if (contextCreatedNotification != null && contextCreatedNotification.poll())
          {
             initializing = false;
 
-            synchronized (syncObject)
-            {
-               setupEyesCount++;
-               context.setupEyes();
-            }
+            context.setupEyes();
 
             scenePoseGizmo = new RDXPose3DGizmo(context.getTeleportFrameIHMCZUp(), context.getTeleportIHMCZUpToIHMCZUpWorld());
             scenePoseGizmo.create(baseUI.getPrimary3DPanel());
             contextInitialized = true;
-            waitGetPosesThreadRunning = true;
-            ThreadTools.startAsDaemon(this::waitOnPoses, getClass().getSimpleName() + "WaitOnPosesThread");
          }
 
          if (contextInitialized)
          {
-            // Waiting for the poses on a thread allows for the rest of the application to keep
-            // cranking faster than VR can do stuff. This also makes the performance graph in Steam
-            // show the correct value and the OpenVR stack work much better.
-            // TODO: This whole thing might have major issues because
-            // there's a delay waiting for the next time this method is called
-            posesReadyThisFrame = posesReady.poll();
-
-            if (!posesReadyThisFrame && !waitingOnPoses)
-            {
-               waitingOnPoses = true;
-               waitOnPosesNotification.set();
-            }
-            else
-            {
-               waitingOnPoses = false;
-            }
+            waitGetPosesFrequencyCalculator.ping();
+            context.waitGetPoses();
```

Comment:
I think the frequency calculator ping should come after the waitGetPoses() method returns... although does the frequency calculator give us any useful information? It'll basically report the UI frequency, right?
```suggestion
            context.waitGetPoses();
            waitGetPosesFrequencyCalculator.ping();
```

Reply:
Yeah, it's useless. Removed.